rfcs/0000-dns-cluster-aware-delegation.md

```diff
@@ -0,0 +1,160 @@
+```
+
+**Example 2.** DNSPolicy using authority delegation with `remote` role
+```yaml
+apiVersion: kuadrant.io/v1
+kind: DNSPolicy
+metadata:
+  name: prod-web-coredns
+spec:
+  authorityDelegation:
+    role: remote
+  targetRef:
+    name: prod-web-istio-coredns
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  providerRefs:
+    - name: dns-provider-credentials-coredns
+```
+
+If the above example policies were created, three DNSRecord resources would exist in the target namespace, with a single DNSRecord resource (the authoritative) appearing as the owner of all endpoints in the target dns provider.
+
+Note: While this concept is intended to work with the multi cluster concept described below, it is also possible to use this on a single cluster to combine several records together. 
+This might be desirable if you have many records and want to reduce the number of API calls to a provider and also the number of owner TXT records that need to be managed.
+
+[//]: # (mnairn: Add a diagram or something to visualise the records being produced  )
+
+## Multi Cluster
+
+The DNS reconciler can be made cluster aware allowing it to process DNSRecord resources on other clusters as well as its own.
+
+Making the controller aware of other clusters is done via specially labeled secret resources containing kubeconfig data that exist in the `kuadrant-system` namespace.
+
+The kubeconfig data should be generated from an appropriately scoped service account that exists on the target cluster.
+
+When a cluster secret is created, the DNS controller on that cluster will automatically detect it and start watching for DNSRecord resources on that external cluster as well as its own.
+
+**Example** Secret containing kubeconfig data
+```yaml
+apiVersion: v1
+kind: Secret
+data:
+  kubeconfig: <kubeconfig data>
+metadata:
+  labels:
+    kuadrant.io/multicluster-kubeconfig: "true"
+  name: kind-kuadrant-local-2
+  namespace: kuadrant-system
+type: Opaque
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## CRD Provider
+
+A new provider implementation that allows a DNSRecord resource to act as a zone record will be created. 
+Endpoints from the source are injected into the target "zone" record using the same plan logic as all other providers. 
+A zone record could be getting updated by many DNSRecords just like any other providers zone.
+
+## Authority Delegation
+
+A new API on the DNSRecord that allows a DNSRecord (and DNSPolicy) to signal that it is delegating authority of publishing to another DNSRecord. 
+Internally uses the CRD provider described above to do this. 
+Assigns a role to the delegation, one of "primary" or "remote" to signal what this particular records role is. 
+Primary records will update/create the authoritative record on the same cluster and supply the provider credentials, remote records only update existing zone records.
+
+## Multicluster
```

Comment:
I think we should add details about the permissions the primary will need in the remote clusters